src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5264,14 +5264,21 @@ private BoundExpression MakeMemberAccessValue(BoundExpression expr, DiagnosticBa
 
                 if (searchExtensionMethodsIfNecessary)
                 {
-                    return new BoundMethodGroup(
+                    var boundMethodGroup = new BoundMethodGroup(
                         node,
                         typeArguments,
                         boundLeft,
                         rightName,
                         lookupResult.Symbols.All(s => s.Kind == SymbolKind.Method) ? lookupResult.Symbols.SelectAsArray(s_toMethodSymbolFunc) : ImmutableArray<MethodSymbol>.Empty,
                         lookupResult,
                         flags);
+
+                    if (!boundMethodGroup.HasErrors && boundMethodGroup.ResultKind == LookupResultKind.Empty && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
```

Comment:
>if (!boundMethodGroup.HasErrors && boundMethodGroup.ResultKind == LookupResultKind.Empty && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument)) [](start = 20, length = 148)

@OmarTawfik Why do we need this special case here only when searchExtensionMethodsIfNecessary is true? It feels like the error should be reported when we do lookup instead.